### PR TITLE
fix(bfdr): roundtrip bug

### PR DESCRIPTION
**Pull request by rpassas**
## Original Issue
'None of the above' fields were not being found by category code + value code, only the value. Constructor was updated to address this. I also had to update `RemoveEntry()` so that the fhirPath would also be checked to remove the correct observations. This was partly why information was getting lost along the way, even when just creating a record.

## Other `roundtrip-ije` Issues Fixed
In testing the the round trip command, I also saw that the fake record had been updated manually and not in alignment with the default one being generated. The fake birth records were regenerated and faulty tests were updated where codes were not correct (e.g., "Osteopath (occupation)" should have been "Osteopath").

Empty values were being inserted during the round trip, causing errors. Setters were updated so that when empty values were passed, empty values were not inserted into the record.

It was also the case that during the round trip there were some type mismatches (e.g. ValueDate used in the software, DateTime used in the IG - [see here](https://mitrecorp.slack.com/archives/GBMV1M7K6/p1710441958445159)). The ValueDate fields were updated to use DateTime and helper functions to parse the DateTime was updated to accommodate. This can be reverted if the IG is updated.

### CODE REVIEW

File: projects/VitalRecord/VitalRecord_util.cs [hunk 1]
```diff
@@ -34,7 +34,15 @@ public abstract partial class VitalRecord
         /// <returns>the first matching resource or element</returns>
         private ITypedElement GetFirstEntryFor(FHIRPath fhirPath)
         {
-            IEnumerable<ITypedElement> matches = Bundle.ToTypedElement().Select(fhirPath.Path);
+            string searchPath;
+            if (fhirPath.Code == NONE_OF_THE_ABOVE)
+            {
+              searchPath = fhirPath.Path + ".where((code as CodeableConcept).coding.code = '"+fhirPath.CategoryCode+"')";
```
[the remark this answers is not in the document]
Comment (saulakravitz) [reply projects/VitalRecord/VitalRecord_util.cs:40]: Seems like we did it backwards.   The category should indicate it is a 'none-of the-above' type, and the code should be specific to the meaning.   This sounds like a defect in the IG.

[the remark this answers is not in the document]
Comment (saulakravitz) [reply projects/VitalRecord/VitalRecord_util.cs:40]: Actually, the FHIRPATH should use Observation.code to get to the class of 'none of the above', and use the value to determine the specific value being set.

**Comment by hadleynet** [comment]
I added some code to the FHIRPath constructor to include the category code in generated paths. This fixes the new test for round tripping but broke another test which incorrectly found none-of-the-above observations for unrelated groupings (only NoObstetricProcedures is explicitly in the FHIR test fixture).

File: projects/VitalRecord/VitalRecord_util.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (hadleynet) [reply projects/VitalRecord/VitalRecord_util.cs]: Right, observation.code = category, observation.value is the code for the specific observation or none-of-the above. Its different for conditions and procedures which have a category attribute where that is used for category and the code is used for the, er, code.

**Review by hadleynet — APPROVED** [review]
Looks good to go from my perspective.

**Review by smacadam — APPROVED** [review]
